flink-runtime-web/web-dashboard/src/app/pages/job/overview/flamegraph/job-overview-drawer-flamegraph.component.ts

```diff
@@ -100,9 +118,38 @@ export class JobOverviewDrawerFlameGraphComponent implements OnInit, OnDestroy {
       );
   }
 
+  private requestRunningSubtasks(): void {
+    this.jobLocalService
+      .jobWithVertexChanges()
+      .pipe(
+        tap(data => (this.selectedVertex = data.vertex)),
+        mergeMap(data => {
+          return this.jobService.loadSubTasks(data.job.jid, data.vertex!.id);
+        }),
+        takeUntil(this.destroy$)
+      )
+      .subscribe(
+        data => {
+          const runningSubtasks = data?.subtasks
```

Comment:
Can `data` be null or undefined in this case? If not, you can skip the optional chaining (`?.`).
 
`runningSubtasks` can be undefined due to the use of optional chaining. This causes the spread operator on line 136 unsafe. Instead, you should wrap it with a IF statement checking whether `runningSubtasks` is not undefined.